### PR TITLE
Added check for XX and T1 counties

### PR DESCRIPTION
**Pull request by harshvitra**

### CODE REVIEW

**Comment by norbertschuler** [comment]
> @norbertschuler maybe we can already map XX T1 to DE in Cloudflare Worker and avoid this patch altogether.

I think every unknown input for country (as always for all parameters ;-)) must be handled properly. We must check if country contains a valid value and otherwise default to e.g. 'DE'.

File: pages/index.tsx [hunk 1]
```diff
@@ -79,7 +79,13 @@ function index({
       setpaymentSetup(paymentSetup);
       setisDirectDonation(isDirectDonation);
     }
-    setcountry(country);
+    // XX is hidden country and T1 is Tor browser
+    if(country === 'XX' || country === 'T1'){
```
[the remark this answers is not in the document]
Comment (norbertschuler) [reply pages/index.tsx:83]: Maybe we must repeat a check against our local country code list here again and default to "DE" if no match was found like at https://github.com/Plant-for-the-Planet-org/pay-with-planet/blob/8e3da3a9d31880766bfe99c5ac83a102ed3dc1ac/pages/index.tsx#L224-L232

**Comment by sagararyal** [comment]
#### 📝Changed routes:


| [**/**](https://donate-with-planet-lczvbswxo-planetapp.vercel.app/) |
|:-:|
|<a href="https://donate-with-planet-lczvbswxo-planetapp.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/TvbB0W8O_QgOnKWk70WiuH24wLriZR-oJ3yNF9Mo7B3vxs6ZaFAywNMo9BxGExAwiZj7g-lyn4b_-Mb2b1mJEQ.png">(view full size)</a> |







Commit 232fead3700e613c0402e87816a692b23c874edf (https://donate-with-planet-lczvbswxo-planetapp.vercel.app).

File: src/Layout/QueryParamContext.tsx [hunk 2]
```diff
@@ -261,7 +259,11 @@ export default function QueryParamProvider({ children }: any) {
       const config: any = await apiRequest(requestParams);
       if (config.data) {
         if (!router.query.country) {
-          setcountry(config.data.country);
+          if (config.data.country == "XX" || config.data.country == "T1") {
```
Comment (harshvitra) [inline src/Layout/QueryParamContext.tsx:262]: Can we check whether the country is available in the array and if it is present we set the country or we set it to DE

```
if (country) { 
	   const found = countriesData.some( 
	     (arrayCountry) => arrayCountry.countryCode?.toUpperCase() === country?.toUpperCase() 
	   ); 
	   if (found) { 
	    setcountry(country);
	   } else{
	    setcountry('DE');
	   }
	 } 
```

Comment (tejassonar) [reply src/Layout/QueryParamContext.tsx:262]: sure

**Review by norbertschuler — APPROVED** [review]
LGFM

**Pull request merged**